### PR TITLE
[feat]:Added News page UI

### PR DESCRIPTION
**Pull request by vpranavkrishna**

### CODE REVIEW

**Comment by satyamurti** [comment]
image
Looks like UI is still the same?

**Comment by satyamurti** [comment]
Refer this UI 
image

**Review by dhruv1294 — APPROVED** [review]

**Review by satyamurti — CHANGES_REQUESTED** [review]
Can you send screen recording of the whole UI once ?

File: lib/pages/news_page.dart [hunk 1]
```diff
@@ -15,20 +17,18 @@ class NewsPage extends StatefulWidget {
...
     context.read<NewsBloc>().add(const GetNews());
-
     context.read<SubscribeCubit>().subscribe(DataStreamType.MARKET_EVENTS);
     _scrollController.addListener(() {
       if (_scrollController.position.atEdge) {
         if (_scrollController.position.pixels != 0) {
-          context.read<NewsBloc>().add(GetMoreNews(10 * i));
-          i++;
+          logger.i(mapMarketEvents.length);
+          context.read<NewsBloc>().add(GetMoreNews(mapMarketEvents.length + 1));
```
Comment (siva2204) [inline lib/pages/news_page.dart:31]: `lastEventId` stands for the latest narket event id thats not fetched yet

lets say we have 20 news in db, the first initial request will send 10 news of id [20,19,18..............12,11], to load the older news the `latestEventId` in this case will be `10` (11 - 1)

this has nothing to with the length, as long as `bool more_exists = 3;` is true in the response, we can fetch older news with the event id

ps : `event id`, `lastEventId` is the primary key in MarketEvents table

File: lib/pages/news_page.dart [hunk 2]
```diff
@@ -49,212 +49,281 @@ class _NewsPageState extends State<NewsPage> {
...
+                          text: text,
+                          imagePath: imagePath,
+                          headline: headline,
+                          dur: dur,
+                        ),
+                      )));
+            } else if (state is SubscriptionToNewsFailed) {
+              return const Center(
+                child: Text(
+                  'Subscription To News Failed to load data \nReason : //',
```
Comment (siva2204) [inline lib/pages/news_page.dart:190]: can use `snackbar` for errors ?

**Review by siva2204 — CHANGES_REQUESTED** [review]
we have append latest news form `marketevents stream` at first in the page

File: lib/pages/news_page.dart (hunk 2 above)
Comment (satyamurti) [reply lib/pages/news_page.dart:190]: It will be shown in center of that section right. I think we should add retry button there.

**Comment by satyamurti** [comment]
What is status of this PR?

**Comment by satyamurti** [comment]
@vpranavkrishna

**Review by satyamurti — APPROVED** [review]
Can you send ss of UI

**Pull request merged**